src/compiler/checker.ts

```diff
@@ -37852,13 +37856,13 @@ namespace ts {
         }
 
         // Keep this up-to-date with the same logic within `getApparentTypeOfContextualType`, since they should behave similarly
-        function findMatchingDiscriminantType(source: Type, target: Type, isRelatedTo: (source: Type, target: Type) => Ternary) {
+        function findMatchingDiscriminantType(source: Type, target: Type, isRelatedTo: (source: Type, target: Type) => Ternary, skipPartial?: boolean) {

```

Comment:
When do you not want to `skipPartial` discriminants?

Reply:
So we report excess property errors on something like
```
type Group = {a: 1, b: 1, field: string, data: string}; | {a: 2, field: string} | {b: 2, data: string};
const x: Group = {a: 1, b: 1, field: "ok", data: "ok", excess: "not ok"};
```
`a` and `b` are both partial discriminants, but taken together, can discriminate to the first member. We have a test expecting this.